src/main/java/seedu/mark/logic/commands/AddCommand.java

```diff
@@ -47,15 +48,15 @@ public AddCommand(Bookmark bookmark) {
 
     @Override
     public CommandResult execute(Model model, Storage storage) throws CommandException {
-        requireNonNull(model);
+        requireAllNonNull(model, storage);
 
         if (model.hasBookmark(toAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_BOOKMARK);
         }
 
         model.addBookmark(toAdd);
         model.applyAllTaggers();
-        model.saveMark();
+        model.saveMark(String.format(MESSAGE_SUCCESS, toAdd));
```

Comment:
is it better to save the CommandResult instead of the message string? because it's quite repetitive

Reply:
Yea that looks better thanks. I'll try to change to that in another PR.